core/src/avm1/globals/sound.rs

```diff
@@ -230,7 +238,10 @@ fn duration<'gc>(
 ) -> Result<Value<'gc>, Error<'gc>> {
     if activation.current_swf_version() >= 6 {
         if let Some(sound_object) = this.as_sound_object() {
-            return Ok(sound_object.duration().into());
+            return Ok(sound_object
+                .sound()
+                .and_then(|s| activation.context.audio.get_sound_duration(s))
+                .map_or(Value::Undefined, |d| d.into()));
```

Comment:
I think we still want to cache the duration, because, for example, loadSound immediately resets it to 0 even if the sound can't be loaded:
```
sound.loadSound("invalid-url.mp3");
trace(sound.duration); // traces 0
```